service/statistics/AnalyticsEvents.js

```diff
@@ -0,0 +1,449 @@
+
+    /**
+     * Properties: value
+     *
+     * Known full event names:
+     * getUserMedia.success
+     * getUserMedia.success.audio
+     * getUserMedia.success.audio.video.180
+     * getUserMedia.success.audio.video.300
+     * getUserMedia.success.audio.video.360
+     * getUserMedia.success.audio.video.720
+     * getUserMedia.success.audio.video.960
+     * getUserMedia.success.audio.video.undefined
+     * getUserMedia.success.desktop
+     * getUserMedia.success.video.180
+     * getUserMedia.success.video.360
+     * getUserMedia.success.video.720
+     * getUserMedia.success.video.960
+     * getUserMedia.success.video.undefined
+     *
+     * TODO: document, reformat
+     */
+    GET_USER_MEDIA_SUCCESS_: 'getUserMedia.success',
+
+    /**
+     * Properties: none
+     *
+     * Known full event names:
+     * getUserMedia.userCancel.extensionInstall
+     *
+     * TODO: document, reformat
+     */
+    GET_USER_MEDIA_USER_CANCEL_: 'getUserMedia.userCancel',
+
+    /**
+     * Properties: value
+     *
+     * The "value" property contains the difference in milliseconds between
+     * the ICE establishment time for the P2P and JVB connections (e.g. a value
+     * of 10 would indicate that the P2P was 10ms slower than JVB).
+     */
+    ICE_ESTABLISHMENT_DURATION_DIFF: 'ice.establishmentDurationDiff',
+
+    /**
+     * Properties: none
+     *
+     * TODO: document
+     * TODO: do we need this in addition to _ICE_CONNECTION_STATE?
+     */
+    ICE_FAILED: 'connection.ice_failed',
+
+    /**
+     * Properties: none
+     *
+     * TODO: document
+     */
+    P2P_ESTABLISHED: 'p2p.established',
+
+    /**
+     * Properties: none
+     *
+     * TODO: document
+     */
+    P2P_FAILED: 'p2p.failed',
+
+    /**
+     * Properties: none
+     *
+     * TODO: document
+     */
+    P2P_SWITCH_TO_JVB: 'p2p.switch_to_jvb',
+
+    /**
+     * Properties: none
+     *
+     * TODO: document
+     */
+    REMOTELY_MUTED: 'conference.remotelyMuted',
+
+    /**
+     * Properties: value
+     *
+     * TODO: document
+     *
+     * The "value" property contains the delay in milliseconds between joining
+     * the MUC and receiving a Jingle session-initiate from Jicofo (but not
+     * P2P).
+     */
+    SESSION_INITIATE: 'session.initiate',
+
+    /**
+     * Properties: value
+     *
+     * TODO: document
+     */
+    SESSION_INITIATE_RECEIVED: 'xmpp.session-initiate',
+
+    /**
+     * Properties: none
+     *
+     * TODO: document
+     */
+    SESSION_TERMINATE: 'session.terminate',
+
+    /**
+     * Properties: none
+     *
+     * TODO: document
+     */
+    SESSION_RESTART: 'session.restart',
+
+    /**
+     * Properties: value
+     *
+     * TODO: document
+     */
+    TRANSPORT_REPLACE_START: 'xmpp.transport-replace.start',
+
+    /**
+     * Properties: value
+     *
+     * TODO: document
+     */
+    TRANSPORT_REPLACE_SUCCESS: 'xmpp.transport-replace.success'
+};
+
+module.exports = AnalyticsEvents;
```

Comment:
Overall I actually like this approach. It's not really a strong commitment to a new implementation but it does the trick for now (and maybe for a long time to come). Where there any alternatives you thought about before choosing this one? When do you suppose those document todos will be done?

Reply:
Not sure about the meaning of "this approach".

If you mean exporting the AnalyticsEvents object, then see the updated commit -- Lyubo suggested exporting the constants individually using "export const", and I went this way because it made the other code cleaner.

If you mean simply having a bunch of constants, then no, I haven't really considered a different approach. I expect that there are libraries and patterns for things like this in javascript, but I don't know of any of them so I just went with the simple thing. This solves most of my issues (being able to trace back where an event comes from, organizing the code in a consistent(-ish) way, and having documentation close to the code).